src/main/java/com/example/bookstore/repository/specification/AuthorSpecProvider.java

```diff
@@ -0,0 +1,21 @@
+package com.example.bookstore.repository.specification;
+
+import com.example.bookstore.model.Book;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "author";
+    }
+
+    @Override
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder) -> root
+                .get("author").in(Arrays.stream(params)

```

Comment:
params already array
```suggestion
                .get("author").in(params

```